internal/api/promote_stage_v1alpha1.go

```diff
@@ -12,24 +13,41 @@ import (
 	svcv1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
 )
 
-// PromoteStage creates a Promotion resource to transition a specified Stage
-// into the state represented by the specified Freight.
+// Promote creates a Promotion resource to transition a specified Stage into the
+// state represented by the specified Freight.
 func (s *server) PromoteStage(
 	ctx context.Context,
 	req *connect.Request[svcv1alpha1.PromoteStageRequest],
 ) (*connect.Response[svcv1alpha1.PromoteStageResponse], error) {
-	if err := validateProjectAndStageNonEmpty(req.Msg.GetProject(), req.Msg.GetName()); err != nil {
-		return nil, err // This already returns a connect.Error
+	project := req.Msg.GetProject()
+	if err := validateFieldNotEmpty("project", project); err != nil {
+		return nil, err
 	}
-	if err := s.validateProjectFn(ctx, req.Msg.GetProject()); err != nil {
-		return nil, err // This already returns a connect.Error
+
+	stageName := req.Msg.GetStage()
+	if err := validateFieldNotEmpty("name", stageName); err != nil {
+		return nil, err
+	}
+
+	freightName := req.Msg.GetFreight()
+	freightAlias := req.Msg.GetFreightAlias()
+	if (freightName == "" && freightAlias == "") || (freightName != "" && freightAlias != "") {
+		return nil, connect.NewError(
+			connect.CodeInvalidArgument,
+			errors.New("exactly one of freightName or freightAlias should not be empty"),
```

Comment:
Nit: I think the "freight name" is called "freight" within `PromoteStageRequest`? Given this, I would make the error message

> exactly one of freight or freightAlias should not be empty

to correctly refer to the field.

Reply:
Nice catch. I believe this accidentally snuck in because this was all extracted from another branch with more extensive changes. Fixing.

Reply:
Fixed in 8f3aea757c1b0ab5531df769ebb5d77eb136b1d2